src/Provider/Capabilities.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * Capabilities provider.
+ *
+ * @package SatisPress
+ * @license GPL-2.0-or-later
+ * @since 0.3.0
+ */
+
+declare ( strict_types = 1 );
+
+namespace SatisPress\Provider;
+
+use Cedaro\WP\Plugin\AbstractHookProvider;
+use SatisPress\Capabilities as Caps;
+use WP_User;
+
+/**
+ * Capabilities provider class.
+ *
+ * @since 0.3.0
+ */
+class Capabilities extends AbstractHookProvider {
+	/**
+	 * Register hooks.
+	 *
+	 * @since 0.3.0
+	 */
+	public function register_hooks() {
+		add_filter( 'map_meta_cap', [ $this, 'map_meta_cap' ], 10, 4 );
+		add_action( 'members_register_cap_groups', [ $this, 'register_members_capability_group' ] );
+		add_action( 'members_register_caps', [ $this, 'register_members_capabilities' ] );
+	}
+
+	/**
+	 * Map meta capabilities to primitive capabilities.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @param array  $caps Returns the user's actual capabilities.
+	 * @param string $cap  Capability name.
+	 * @return array
+	 */
+	public function map_meta_cap( array $caps, string $cap ): array {
+		switch ( $cap ) {
+			case Caps::DOWNLOAD_PACKAGE :
+				$caps = [ Caps::DOWNLOAD_PACKAGES ];
+				break;
+
+			case Caps::VIEW_PACKAGE :
+				$caps = [ Caps::VIEW_PACKAGES ];
+				break;
+		}
+
+		return $caps;
+	}
+
+	/**
+	 * Register a capability group for the Members plugin.
```

Comment:
While it might be popular, this doesn't seem like it could scale to support other Membership-related plugins, without breaking the SRP.

Could support for other plugins be moved to their own Provider classes, rather than being bundled by feature?

Adding support for a new plugin then means only touching one class, instead of touching Capabilities, Authentication and whatever other classes. These classes could then only ever be instantiated if that plugin was detected to be active.

Reply:
I can't slip nothing by you! I do typically create a separate provider for each plugin integration, but added those methods here for convenience. I'll go ahead and split them out.

Reply:
Sorry! Reviewing code is as much about asking questions to validate my own understanding.